console/src/main/scripts/plugins/metrics/ts/directives/datasources/dataSourceController.ts

```diff
@@ -0,0 +1,305 @@
+            };
+          })
+        );
+      }
+      if (!this.skipChartData[this.datasource.id + '_In Use Count']) {
+        availPromises.push(this.getInUsePromise()
+          .then((data) => {
+            return {
+              key: 'In Use',
+              color: AppServerDatasourcesDetailsController.IN_USE_COLOR,
+              values: MetricsService.formatBucketedChartOutput(data)
+            };
+          })
+        );
+      }
+      if (!this.skipChartData[this.datasource.id + '_Timed Out']) {
+        availPromises.push(this.getTimedOutPromise()
+          .then((data) => {
+            return {
+              key: 'Timed Out',
+              color: AppServerDatasourcesDetailsController.TIMED_OUT_COLOR,
+              values: MetricsService.formatBucketedChartOutput(data)
+            };
+          })
+        );
+      }
+      return availPromises;
+    }
+
+    /**
+     * Method for creating chart data from raw IChartDataPoint data.
+     * @returns {ng.IPromise<IMultipleChartData>[]} formatted Data for multi-line response chart.
+     */
+    public getResponseChartPromises():ng.IPromise<IMultipleChartData>[] {
+      let responsePromises:ng.IPromise<IMultipleChartData>[] = [];
+      if (!this.skipChartData[this.datasource.id + '_Average Get Time']) {
+        responsePromises.push(this.getAvgGetTimePromise()
+          .then((data) => {
+            return {
+              key: 'Wait Time (Avg.)',
+              color: AppServerDatasourcesDetailsController.WAIT_COLOR,
+              values: MetricsService.formatBucketedChartOutput(data)
+            };
+          })
+        );
+      }
+      if (!this.skipChartData[this.datasource.id + '_Average Creation Time']) {
+        responsePromises.push(this.getAvgCreateTimePromise()
+          .then((data) => {
+            return {
+              key: 'Creation Time (Avg.)',
+              color: AppServerDatasourcesDetailsController.CREATION_COLOR,
+              values: MetricsService.formatBucketedChartOutput(data)
+            };
+          })
+        );
+      }
+      return responsePromises;
+    }
+
+    /**
+     * Method for constructing promise with data for chart of Available count in Datasource pool.
+     * @returns {ng.IPromise<IChartDataPoint[]>} constructed promise with data point for available chart.
+     */
+    public getAvailablePromise():ng.IPromise<IChartDataPoint[]> {
+      return this.MetricsService.retrieveGaugeMetrics(
+        this.currentPersona.id,
+        MetricsService.getMetricId('M', this.feedId, this.datasource.id, 'Datasource Pool Metrics~Available Count'),
+        this.startTimeStamp,
+        this.endTimeStamp,
+        60);
+    }
+
+    /**
+     * Method for constructing promise with data for chart of Use Count in Datasource pool.
+     * @returns {ng.IPromise<IChartDataPoint[]>} constructed promise with data point for available chart.
+     */
+    public getInUsePromise():ng.IPromise<any> {
```

Comment:
Let's not use <any> here since we know what we are returning ng.IPromise<IChartDataPoint[]>